pkg/bench/ring_check.go

```diff
@@ -53,15 +54,25 @@ func NewRingChecker(id string, instanceName string, cfg RingCheckConfig, workloa
 		logger:   logger,
 		workload: workload,
 	}
-	cfg.MemberlistKV.MetricsRegisterer = prometheus.DefaultRegisterer
+	reg := prometheus.DefaultRegisterer
+	cfg.MemberlistKV.MetricsRegisterer = reg
 	cfg.MemberlistKV.Codecs = []codec.Codec{
 		ring.GetCodec(),
 	}
-	r.MemberlistKV = memberlist.NewKVInitService(&cfg.MemberlistKV, logger)
+
+	dnsProviderReg := prometheus.WrapRegistererWithPrefix(
```

Comment:
Does this change the names of the existing metrics?

Reply:
The [original](https://github.com/cortexproject/cortex/blob/v1.9.0/pkg/ring/kv/memberlist/kv_init_service.go#L42) has no Prometheus metrix. This is just copy&paste from cortex. Should I use `reg` instead?

Reply:
I have no idea - the authors will probably have to weigh in on this

Reply:
@replay you fingerprints are on the `ring_check.go` and @aknuds1 you've migrated Cortex to dskit for `NewKVInitService`. Should we use a prefix or just go with the `prometheus.DefaultRegisterer`here?

Reply:
@jeschkies in order to retain the `cortex_` metrics prefix, you have to configure the same prefix here yes.

Reply:
I'm not very familiar with this code, I show up in the git blame only because I renamed some symbols. I'd like to defer to somebody who is more familiar

Reply:
> you do want to retain the metrics prefix from before porting to dskit right?

Yes. So I think we should go with this solution then.

@replay thanks for taking a look nonetheless. 

@dannykopping I think this is good to go then :slightly_smiling_face:

Reply:
I believe @gotjosh is one of the main authors here - it'd be good to get their opinion on this